docs/concepts/models.md

```diff
@@ -181,36 +208,22 @@ print(m.model_dump())
 """
 ```
 
-For self-referencing models, see [postponed annotations](postponed_annotations.md#self-referencing-or-recursive-models).
-
-!!! note
-    When defining your models, watch out for naming collisions between your field name and its type, a previously defined model, or an imported library.
-
-    For example, the following would yield a validation error:
-    ```py test="skip"
-    from typing import Optional
-
-    from pydantic import BaseModel
-
+Self-referencing models are supported. For more details, see [postponed annotations](postponed_annotations.md#self-referencing-or-recursive-models).
 
-    class Boo(BaseModel):
-        int: Optional[int] = None
+## Rebuilding the model
```

Comment:
I think this should be "Rebuilding model schema", as we don't rebuild the model entirely

Reply:
I wanted to avoid mentioning the concept of "schema", which isn't introduced at this point in the docs. But "Rebuilding the model" makes it like the model is completely redefined, which isn't quite true. I'll adapt the description to have a quick explanation of what a core schema is.